File: MatlabOctaveCodeSnippets/snip_signals_wavread.m
%% Get samples using fread and skipping header:
fp=fopen('c:/temp/myvoice.wav','rb'); %open for reading in binary
x=fread(fp,Inf,'int16'); %read all samples as signed 16-bits
fclose(fp);  %close the file
x(1:22)=[]; %eliminate the 44-bytes header of a Microsoft RIFF file
%% Now using readwav in folder MatlabOctaveThirdPartyFunctions:
[x2,Fs,wmode,fidx]=readwav('c:/temp/myvoice.wav','r'); % raw samples
b = fidx (7); % num of bits per sample
x2=double(x2); %convert integer to double for easier manipulation
isequal(x,x2) %result must be 1, indicating vectors are identical
%% Now using Matlab's audioread (not available in Octave):
%[x3, fs] = audioread('c:/temp/myvoice.wav', 'native');
%isequal(x,x3) %result must be 1, indicating vectors are identical

File: PythonCodeSnippets/snip_signals_wavread.py
import numpy as np
import wave

# Specify a WAV file_name
file_name = "myvoice.wav"

# Now we open the WAV file
with open(file_name, "rb") as audio_file:  # open for reading in binary
    frames = audio_file.read()

x = np.frombuffer(frames, dtype=np.int16)  # read all samples as signed 16-bits
x = x[22:]  # eliminate the 44-bytes header

# Get Raw samples
with wave.open(file_name, "r") as wf:
    x2 = np.frombuffer(wf.readframes(wf.getnframes()), dtype=np.int16)
    Fs = wf.getframerate()
    sample_width = wf.getsampwidth()  # Returns sample width in bytes.
    wmode = (
        wf.getnchannels()
    )  # Returns number of audio channels (1 for mono, 2 for stereo).

# Get the number of bits per sample
b = sample_width * 8  # Conversion to bits: 1 byte = 8 bits

x2 = x2.astype(np.double)  # convert integer to double for easier manipulation
result = np.array_equal(x, x2)
print(result)  # result is True, indicating vectors are identical
